runtime/runtime.m

```diff
@@ -1031,11 +1031,8 @@ -(void) xamarinSetGCHandle: (int) gc_handle;
 {
 	MTClassMap *mapa = (MTClassMap *) a;
 	MTClassMap *mapb = (MTClassMap *) b;
-	if (mapa->handle == mapb->handle)
-		return 0;
-	if ((intptr_t) mapa->handle < (intptr_t) mapb->handle)
-		return -1;
-	return 1;
+
+	return (intptr_t)mapa->handle - (intptr_t)mapb->handle;
```

Comment:
The question would be what is the comparison being used for. If we have the overflow issue, but we are only comparing for equality and not for sorting,  we don't care if it is positive or not.

On the other hand, if we are sorting, we will get in a bit of a pickle. In this case, we are using it for sorting, so overflowing is indeed a possibility. I personally, would avoid the change since I do not know use case for the runtime. But I trust @spouliot and @rolfbjarne in their +1.

Reply:
Yeah, I opened the PR thinking it was a good optimization, then I doubted I knew anything.

My reasoning for disregarding the overflow is that if you have a static registrar that crosses 32bit boundary as values, then you have bigger problems. Still, you never konw.

Reply:
@Therzok that is a very valid point, we could make the return type to be bigger than an int since adding the check for the overflow will be undoing the optimization done with the operation. 

Maybe something like this?
```c
int cmp(int x, int y) {
    return x = x - y, ( x >> 31 ) | !!x;
}
```

Reply:
This is so clever ❤️ @mandel-macaque 🎉